### PR TITLE
Feat: (491)Added rate feeds for CELOPHP & PHPUSD

### PR DESCRIPTION
**Pull request by RyRy79261**
### Description

We're adding metrics for our first relayed oracle feeds CELO/PHP and PHP/USD.

- [x] Added new metric for `SortedOracles.isOldestReportExpired()` (because `numRates` doesn't make sense for chainlink feeds as they're all gonna be a 1/1 config in SortedOracles)
- [x] Refactored `Metric.parse()` logic to be easier to customize per metric
- [x] Wrote small `deploy.sh` script callable via `npm run deploy` that will automatically switch to the `mento-prod` gcloud project in case it was forgotten to do manually

### CODE REVIEW

**Review by philbow61 — CHANGES_REQUESTED** [review]
I think we only need the tradingMode for the USDPHP rate the other ones don't make sense with the chainlink setup

**Review by philbow61 — COMMENTED** [review]
I think you kept the wrong one we only want to have the TradingMode and only for PHPUSD.
The CELOPHP one can be removed completely

File: contracts/OracleHelper.sol [hunk 1]
```diff
@@ -1,54 +1,58 @@
...
+    address public immutable sortedOracles;
+
+    constructor(address _sortedOracles) {
+        sortedOracles = _sortedOracles;
     }
 
-    return (intoUint256(maxDiff), 1e18);
-  }
+    /**
+     * @notice Get the deviation of the median rate of a rate feed from the mean of all rates.
```
Comment (bayological) [inline contracts/OracleHelper.sol:31]: nit: I'd go all the way with this and add a param tag to briefly clarify what 'token' is in the function sig.

File: src/metric.ts [hunk 2]
```diff
@@ -58,26 +58,37 @@ export class Metric {
...
-      return Number(parsed);
-    } else {
-      const [numerator, denominator] = output as [bigint, bigint];
-      if (denominator === BigInt(0)) {
-        return 0;
-      }
-      const precision = 1e6;
-      const value = (numerator * BigInt(precision)) / denominator;
-      if (value > Number.MAX_SAFE_INTEGER) {
+  parse(output: any, functionName: string): number {
```
Comment (bayological) [inline src/metric.ts:61]: Let's actually add some tests for this function. This is a significant change.

**Review by bayological — CHANGES_REQUESTED** [review]

File: config.local.yaml [hunk 3]
```diff
@@ -8,7 +8,7 @@ global:
     EUROCXOF: '0xed35e46b095197da30ddffa5b91d386886d5ce0d'
     KESUSD: '0xbAcEE37d31b9f022Ef5d232B9fD53F05a531c169'
     USDTUSD: '0xE06C10C63377cD098b589c0b90314bFb55751558'
-
```
Comment (philbow61) [inline config.local.yaml:11]: the CELOPHP rateFeedID is missing here

File: config.yaml [hunk 4]
```diff
@@ -8,7 +8,7 @@ global:
     EUROCXOF: '0xed35e46b095197da30ddffa5b91d386886d5ce0d'
     KESUSD: '0xbAcEE37d31b9f022Ef5d232B9fD53F05a531c169'
     USDTUSD: '0xE06C10C63377cD098b589c0b90314bFb55751558'
-
```
Comment (philbow61) [inline config.yaml:11]: CELOPHP rateFeedID is missing here

**Review by philbow61 — CHANGES_REQUESTED** [review]

File: src/metric.ts (hunk 2 above)
Comment (RyRy79261) [reply src/metric.ts:61]: @bayological What would the test case be and what is a normal value for this?

File: config.yaml (hunk 4 above)
Comment (RyRy79261) [reply config.yaml:11]: @philbow61 You specified to remove it

Comment (philbow61) [reply config.yaml:11]: @RyRy79261  Me and Philip II created a new metric that does make sense with the chainlink setup.
It's an isOldestReportExpired() call to sortedOracles. that's why we need to add that rateFeed again. 
You can see that we are also referencing it in the definition of that isOldestReportExpired() metric.

File: config.local.yaml
[the remark this answers is not in the document]
Comment (chapati23) [reply config.local.yaml:1]: yup, they're superfluous according to yaml lint

[the remark this answers is not in the document]
Comment (chapati23) [reply config.local.yaml:1]: deployment worked fine w/o them fwiw

File: src/metric.ts (hunk 2 above)
Comment (chapati23) [reply src/metric.ts:61]: i'm all for testing but tbf this project has 0 tests atm and the function was even more opaque before this change. 
would argue for creating an issue and adding more than just this 1 test in another PR.

Comment (chapati23) [reply src/metric.ts:61]: yeah, just tried adding some basic test and was confronted with a typescript x eslint config issue rabbit hole that i don't think we should go down in this PR

File: config.yaml (hunk 4 above)
Comment (chapati23) [reply config.yaml:11]: i re-added it, got lost in the back and forth

**Review by philbow61 — APPROVED** [review]
LGTM 👍

File: src/metric.ts (hunk 2 above)
Comment (bayological) [reply src/metric.ts:61]: > @bayological What would the test case be and what is a normal value for this?

You could cover each case in the switch statement including the edge cases and error scenarios. There isn't a single 'normal value' for this.

**Review by bayological — APPROVED** [review]
Nice one 👍

**Review by chapati23 — APPROVED** [review]
well done, incredible code here
